website/src/components/Chat/ChatSection.tsx

```diff
@@ -0,0 +1,212 @@
+  Select,
+  Slider,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+  Stack,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { Settings } from "lucide-react";
+import { useTranslation } from "next-i18next";
+import { useCallback, useState } from "react";
+import { Controller, useForm, UseFormReturn } from "react-hook-form";
+import { get } from "src/lib/api";
+import { API_ROUTES } from "src/lib/routes";
+import { ModelInfo, WorkParametersInput } from "src/types/Chat";
+import useSWRImmutable from "swr/immutable";
+import { StrictOmit } from "ts-essentials";
+
+import { ChatConversation } from "./ChatConversation";
+
+type UseChatConfigForm = UseFormReturn<WorkParametersInput>;
+
+export const ChatSection = ({ chatId }: { chatId: string }) => {
+  const { control, getValues, register, setValue } = useForm<WorkParametersInput>({
+    defaultValues: {
+      temperature: 0.7,
+      top_p: 0.5,
+      top_k: 1.2,
+      repetition_penalty: 1.2,
+      max_new_tokens: 256,
+    },
+  });
+
+  return (
+    <>
+      <ChatConversation chatId={chatId} getConfigValues={getValues} />
+      <ChatConfigDrawer control={control} register={register} setValue={setValue}></ChatConfigDrawer>
+    </>
+  );
+};
+
+type ChatConfigDrawerProps = Pick<UseChatConfigForm, "control" | "register" | "setValue">;
+
+const ChatConfigDrawer = ({ control, register, setValue }: ChatConfigDrawerProps) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const { data, isLoading } = useSWRImmutable<ModelInfo[]>(API_ROUTES.GET_CHAT_MODELS, get, {
+    onSuccess(models) {
+      const model = models[0];
+
+      model && setValue("model_name", model.name);
+    },
+  });
+  const { t } = useTranslation("chat");
+  return (
+    <>
+      <Portal>
+        <IconButton
+          position="absolute"
+          style={{
+            insetInlineEnd: `52px`,
+          }}
+          bottom={10}
+          aria-label={t("chat_config.title")}
+          icon={<Settings></Settings>}
+          size="lg"
+          borderRadius="2xl"
+          colorScheme="blue"
+          onClick={onOpen}
+        ></IconButton>
+      </Portal>
+      <Drawer placement="right" onClose={onClose} isOpen={isOpen}>
+        <DrawerOverlay />
+        <DrawerContent>
+          <DrawerCloseButton />
+          <DrawerHeader borderBottomWidth="1px">{t("chat_config.title")}</DrawerHeader>
+          <DrawerBody>
+            {isLoading && !data && <CircularProgress></CircularProgress>}
+            {data && <ChatConfigForm modelInfos={data} control={control} register={register}></ChatConfigForm>}
+          </DrawerBody>
+        </DrawerContent>
+      </Drawer>
+    </>
+  );
+};
+
+const sliderItems = [
+  {
+    key: "temperature",
+  },
+  {
+    key: "max_new_tokens",
+    max: 1000,
+    step: 1,
+    min: 1,
+  },
+  {
+    key: "top_p",
+  },
+  {
+    key: "repetition_penalty",
+    min: 1,
+    max: 2,
+  },
+  {
+    key: "top_k",
+    min: 1,
+    max: 2,
+  },
```

Comment:
I'm not sure about the range of these parameters. But for `repetition_penalty` and `top_k` it should be greater than 1 otherwise the inference backend will crash

Reply:
cc @yk

Reply:
`temperature`: Temperature of 1.0 is neutral, lowering this makes the sampling greedier, maybe range could be 0-4
`top_p`: is something in the interval 0-1, percentage of probability mass to consider
`top_k`: is a number of the top k logits to consider, can be 1 to number of vocabulary (usually something like 32k), especially smaller values like 50 are interesting, maybe use exponential transfer function of slider value x -> `Math.pow(32000, x)` for 0 <= x <= 1...
`repetition_penalty`: 1.0 means no penalty, paper that invents this recommends: "We find that using a greedy sampling and θ ≈ 1.2 yields a good balance between truthful generation and lack of repetition." .. maybe range could also be 1-3 (not sure if someone uses higher values)